core/storage/suggestion/gae_models_test.py

```diff
@@ -2994,3 +2994,1639 @@ def test_get_model_association_to_user(self) -> None:
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationSubmitterTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=0,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingSubmissions'],
+            topic_ids=None,
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_2')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 1)
+
+        # Check for topic filter and non-performance sort order.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationSubmitterTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=1,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingPerformance'],
+            topic_ids=['topic1', 'topic2'],
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_3')
+        self.assertFalse(more)
+        self.assertEqual(next_offset, 2)
+
+        # Check for language filter.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationSubmitterTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=0,
+            sort_by=None,
+            topic_ids=None,
+            last_activity=None,
+            language_code='hi'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_4')
+        self.assertFalse(more)
+        self.assertEqual(next_offset, 1)
+
+        # Check for decreasing last activity sort order.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationSubmitterTotalContributionStatsModel
+            .fetch_page(
+            page_size=2,
+            offset=0,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'DecreasingLastActivity'],
+            topic_ids=None,
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_3')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 2)
+
+        # Check for increasing last activity sort order.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationSubmitterTotalContributionStatsModel
+            .fetch_page(
+            page_size=2,
+            offset=0,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingLastActivity'],
+            topic_ids=None,
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_1')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 2)
+
+        # Check for last activity under 7 days.
```

Comment:
I suggest creating a separate test for filters since they are conceptually different from sorts.

Reply:
I have covered and tested every sort and filter we have implemented. Also every branch in the code. I am little confused here wdym by a separete test, do you mean we should create a different test class?

Reply:
Similar to the other comment I left, I would suggest having different test methods (not classes) -- one for testing sorting, one for testing filtering, and one for testing the combo. Also, name your test methods clearly -- not just "test_method_name". See https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests for guidance.

It's generally not a good idea to have N different behavioural tests in a single method.

Reply:
I have splitted tests into four tests. 
sorting
filtering
sorting and filtering
pagination.